backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -849,6 +853,35 @@ public Set<Observation> convertToAOESymptomObservation(
     return observations;
   }
 
+  public Set<Observation> convertToAOEGenderOfSexualPartnersObservation(
+      List<String> sexualPartners) {
+    HashSet<Observation> observations = new LinkedHashSet<>();
+
+    for (String sexualPartner : sexualPartners) {
+      Map<String, String> genderMap = getGenderCodeMap();
+      String genderSNOMED = genderMap.get(sexualPartner);
+
+      Map<String, String> genderDisplayMap = getGenderDisplayMap();
+      String genderStatusDisplay = genderDisplayMap.get(genderSNOMED);
+
+      CodeableConcept genderOfSexualPartnerStatusCode =
+          createSimpleReportConcept(
+              SIMPLE_REPORT_GENDER_OF_SEXUAL_PARTNERS,
+              "What is the gender of their sexual partners",
+              "What is the gender of their sexual partners");
+
+      CodeableConcept genderOfSexualPartnerValueCode =
+          createSNOMEDConcept(genderSNOMED, genderStatusDisplay);
+
+      observations.add(
+          createAOEObservation(
+              uuidGenerator.randomUUID() + SIMPLE_REPORT_GENDER_OF_SEXUAL_PARTNERS,
```

Comment:
iI there a reason why we need the second half of`+ SIMPLE_REPORT_GENDER_OF_SEXUAL_PARTNERS` here?  If I'm understanding correctly, it looks like we take this string and generate a UUID from that as a seed, so the constant shouldn't impact the output UUID since the random part in the generator is what's producing the variation.

Reply:
II wasn't quite sure if it was needed either. However, it is how every other convert to AOE observation function does it so I decided to follow the already established pattern. For an example of this patern see `[here](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java#L855-L864)`

Reply:
@emyl3 might know why we set the unique IDs to contain this value

Reply:
Other examples use the test eventid + `SOME_FIELD_NAME`.  I'm guessing that for situations where there are multiple values of that type, that would result in the same UUID for all of them (i.e. collisions, very bad).  

As to why we're using `nameUUIDFromBytes` instead of just generating a UUID? 🤷

Reply:
I think I initially implemented this based on what I saw from the example Dan shared "test eventid + `SOME_FIELD_NAME`) 

I think it makes sense to just generate a UUID and we can clean up other uses of this in other methods 😅